pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -0,0 +1,255 @@
+		kubeInformersForTargetNamespace.Core().V1().Nodes().Informer(),
+	).WithSync(c.sync).ToController("GuardController", eventRecorder)
+}
+
+func getInstallerPodImageFromEnv() string {
+	return os.Getenv("OPERATOR_IMAGE")
+}
+
+func getGuardPodName(prefix, nodeName string) string {
+	return fmt.Sprintf("%s-guard-%s", prefix, nodeName)
+}
+
+func getGuardPDBName(prefix string) string {
+	return fmt.Sprintf("%s-guard-pdb", prefix)
+}
+
+func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	klog.V(5).Info("Syncing guards")
+
+	errs := []error{}
+	shouldDelete := false
+	if c.deleteConditionalFunc != nil && c.deleteConditionalFunc() {
+		shouldDelete = true
+	}
+
+	shouldCreate := false
+	if c.createConditionalFunc != nil && c.createConditionalFunc() {
+		shouldCreate = true
+	}
+
+	switch {
+	case shouldCreate && shouldDelete:
+		errs = append(errs, fmt.Errorf("can not have both create and deletion conditionals evaluated to true at the same time"))
+	case shouldDelete:
+		pdb := resourceread.ReadPodDisruptionBudgetV1OrDie(bindata.MustAsset(filepath.Join("pkg/operator/staticpod/controller/guard", "manifests/pdb.yaml")))
+		pdb.ObjectMeta.Name = getGuardPDBName(c.podResourcePrefix)
+		pdb.ObjectMeta.Namespace = c.targetNamespace
+		_, _, err := resourceapply.DeletePodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
+		if err != nil {
+			klog.Errorf("Unable to delete PodDisruptionBudget: %v", err)
+			errs = append(errs, err)
+		}
+
+		pods, err := c.podGetter.Pods(c.targetNamespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{"app": "guard"}).String(),
+		})
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			for _, pod := range pods.Items {
+				_, _, err = resourceapply.DeletePod(ctx, c.podGetter, syncCtx.Recorder(), &pod)
+				if err != nil {
+					klog.Errorf("Unable to delete Pod: %v", err)
+					errs = append(errs, err)
+				}
+			}
+		}
+	case shouldCreate:
+		selector, err := labels.NewRequirement("node-role.kubernetes.io/master", selection.Equals, []string{""})
+		if err != nil {
+			panic(err)
+		}
+		nodes, err := c.nodeLister.List(labels.NewSelector().Add(*selector))
+		if err != nil {
+			return err
+		}
+
+		pods, err := c.podGetter.Pods(c.targetNamespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{"app": c.podResourcePrefix}).String(),
+		})
+		if err != nil {
+			return err
+		}
+
+		klog.V(5).Infof("Rendering guard pdb")
+
+		pdb := resourceread.ReadPodDisruptionBudgetV1OrDie(bindata.MustAsset(filepath.Join("pkg/operator/staticpod/controller/guard", "manifests/pdb.yaml")))
+		pdb.ObjectMeta.Name = getGuardPDBName(c.podResourcePrefix)
+		pdb.ObjectMeta.Namespace = c.targetNamespace
+		if len(nodes) > 1 {
+			minAvailable := intstr.FromInt(len(nodes) - 1)
+			pdb.Spec.MinAvailable = &minAvailable
+		}
+
+		_, _, err = resourceapply.ApplyPodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
```

Comment:
The PDB portion of this could be applied using our static resource controller and a `WithConditionalResource`, right?

Reply:
> The PDB portion of this could be applied using our static resource controller and a `WithConditionalResource`, right?

I see, it's dynamic. nevermind then.